### ISSUE TITLE
Kismatic function to list storage volumes

### ISSUE THREAD

**Issue by dasmb** [apprenda/kismatic#204]
One component of storage management is understanding what data is in use by which workloads.

`kismatic storage list-volumes -f kismatic-cluster.yaml`

will list all PVs on a cluster, along with
 - whether they've been claimed or not
 - Their size

& I dunno -- the workloads using them, details about how the volume is distributed around a Gluster cluster.

**Comment by dkoshkin** [comment]
I see a few approaches to do this:

1. Pick up a dependency on`kubeconfig` file and kubectl and run `kubectl` from the controller machine
2. Have a dependency on the plan file, and proxy ssh commands via one of the masters
3. Have a dependency on the plan file, and run `kubectl` with Ansible
4. Some other method anyone can think of
5. Do not do anything and just document the use of `kubectl get pv`

Eventually I see this command as linking the different layers: gluster volumes and NFS -> PV -> PVC -> pods consuming the PVC but this is complicated and I dont see an necessary for v1 of storage or something that should hold back that release.

If someone sees a different approach lets discuss here.

**Comment by dkoshkin** [comment]
```
$ kubectl get pvc
NAME             STATUS    VOLUME                 CAPACITY   ACCESSMODES   AGE
myclaim-1        Bound     kismatic-test-volume   1Gi        RWX           6m
```
```
$ kubectl get pv
NAME      LABELS       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS       CLAIM               REASON     AGE
pv0001    type=local   1Gi        RWX           Retain          Available                                   10m
pv0002    type=local   1Gi        RWX           Retain          Bound        default/myclaim-1              10m
```
```
$ kismatic volume list

NAME      CAPACITY   REPLICA  DISTRIBUTION  STATUS         CLAIM               BRICK(S)
pv0001    1Gi        2        2             Available                          node1:/data/pv0001,node2:/data/pv0001,node3:/data/pv0001,node4:/data/pv0001
pv0002    1Gi        1        1             Bound          default/myclaim-1   node1:/data/pv0002,node2:/data/pv0002
```

The idea here is we are not trying to replace the existing `get pv` and `get pvc` commands, instead to expose the GlusterFS details. 

I can see an argument for `ACCESSMODES` and `RECLAIMPOLICY` to also be included but its less valuable.

**Comment by dkoshkin** [comment]
[XML output from Gluster]( https://gist.github.com/dkoshkin/f0b21c892a64d17928a0198f391899f3)

``` xml
...
    <volumes>
      <volume>
       <name>storage01</name>
        <distCount>2</distCount>
        <replicaCount>2</replicaCount>
        <bricks>
          <brick uuid="31de875a-23f6-438b-aa47-246bef2a0e0b">ip-10-0-3-235:/data/storage01<name>ip-10-0-3-235:/data/storage01</name><hostUuid>31de875a-23f6-438b-aa47-246bef2a0e0b</hostUuid><isArbiter>0</isArbiter></brick>
          <brick uuid="34700865-a17d-4bb0-b293-949cc97c68d9">ip-10-0-3-201:/data/storage01<name>ip-10-0-3-201:/data/storage01</name><hostUuid>34700865-a17d-4bb0-b293-949cc97c68d9</hostUuid><isArbiter>0</isArbiter></brick>
          <brick uuid="9a4ff890-a4f3-4e44-9e0d-457a3ea61234">ip-10-0-3-16:/data/storage01<name>ip-10-0-3-16:/data/storage01</name><hostUuid>9a4ff890-a4f3-4e44-9e0d-457a3ea61234</hostUuid><isArbiter>0</isArbiter></brick>
          <brick uuid="f43f08aa-d71f-4c2c-8b91-f1e9db8087a7">ip-10-0-3-176:/data/storage01<name>ip-10-0-3-176:/data/storage01</name><hostUuid>f43f08aa-d71f-4c2c-8b91-f1e9db8087a7</hostUuid><isArbiter>0</isArbiter></brick>
        </bricks>
     </volume>
   </volumes>
...
```

To get `AVAILABLE_SPACE` we would need to loop through all of the volumes and pick up the `<avail_space>` tag

**Comment by alexbrand** [comment]
I worry that we are displaying too much data on a single row. Have you experimented with other ways of displaying the data?

**Comment by dkoshkin** [comment]
I think you are right, this is what it looks like

```
➜  out git:(master) ✗ ./kismatic volume list
NAME                   CAPACITY   REPLICA   DISTRIBUTION   BRICK(S)                                                                                                                 STATUS      CLAIM               POD(S)
kismatic-test-volume   1.00GB     1         1              ip-10-0-3-211:/data/kismatic-test-volume                                                                                 Bound       default/claim1      default/mypod1,default/mypod3
storage01              2.00GB     2         1              ip-10-0-3-211:/data/storage01,ip-10-0-3-245:/data/storage01                                                              Available
storage02              2.00GB     1         2              ip-10-0-3-211:/data/storage02,ip-10-0-3-162:/data/storage02                                                              Available
storage03              1.00GB     2         2              ip-10-0-3-245:/data/storage03,ip-10-0-3-162:/data/storage03,ip-10-0-3-211:/data/storage03,ip-10-0-3-48:/data/storage03   Bound       default/claim2      default/mypod2,default/mypod4
storage04              2.00GB     4         1              ip-10-0-3-245:/data/storage04,ip-10-0-3-211:/data/storage04,ip-10-0-3-162:/data/storage04,ip-10-0-3-48:/data/storage04   Bound       default/my-claim3
storage05              2.00GB     1         4              ip-10-0-3-211:/data/storage05,ip-10-0-3-162:/data/storage05,ip-10-0-3-245:/data/storage05,ip-10-0-3-48:/data/storage05   Available
storage06              2.00GB     1         3              ip-10-0-3-211:/data/storage06,ip-10-0-3-245:/data/storage06,ip-10-0-3-162:/data/storage06                                Available
storage07              2.00GB     3         1              ip-10-0-3-211:/data/storage07,ip-10-0-3-245:/data/storage07,ip-10-0-3-162:/data/storage07                                Available
```

**Comment by dkoshkin** [comment]
Here is another alternative
```
➜  out git:(master) ✗ ./kismatic volume list
Name: kismatic-test-volume
Capacity: 1.00GB
Available: 1.00GB
Available: 1
Available: 1
Bricks: ip-10-0-3-211:/data/kismatic-test-volume
Status: Bound
Claims: default/claim1
Pods: default/mypod1,default/mypod3

Name: storage01
Capacity: 2.00GB
Available: 2.00GB
Available: 2
Available: 1
Bricks: ip-10-0-3-211:/data/storage01,ip-10-0-3-245:/data/storage01
Status: Available
Claims:
Pods:

Name: storage02
Capacity: 2.00GB
Available: 2.00GB
Available: 1
Available: 2
Bricks: ip-10-0-3-211:/data/storage02,ip-10-0-3-162:/data/storage02
Status: Available
Claims:
Pods:

Name: storage03
Capacity: 1.00GB
Available: 1.00GB
Available: 2
Available: 2
Bricks: ip-10-0-3-245:/data/storage03,ip-10-0-3-162:/data/storage03,ip-10-0-3-211:/data/storage03,ip-10-0-3-48:/data/storage03
Status: Bound
Claims: default/claim2
Pods: default/mypod2,default/mypod4

Name: storage04
Capacity: 2.00GB
Available: 2.00GB
Available: 4
Available: 1
Bricks: ip-10-0-3-245:/data/storage04,ip-10-0-3-211:/data/storage04,ip-10-0-3-162:/data/storage04,ip-10-0-3-48:/data/storage04
Status: Bound
Claims: default/my-claim3
Pods:

Name: storage05
Capacity: 2.00GB
Available: 2.00GB
Available: 1
Available: 4
Bricks: ip-10-0-3-211:/data/storage05,ip-10-0-3-162:/data/storage05,ip-10-0-3-245:/data/storage05,ip-10-0-3-48:/data/storage05
Status: Available
Claims:
Pods:

Name: storage06
Capacity: 2.00GB
Available: 2.00GB
Available: 1
Available: 3
Bricks: ip-10-0-3-211:/data/storage06,ip-10-0-3-245:/data/storage06,ip-10-0-3-162:/data/storage06
Status: Available
Claims:
Pods:

Name: storage07
Capacity: 2.00GB
Available: 2.00GB
Available: 3
Available: 1
Bricks: ip-10-0-3-211:/data/storage07,ip-10-0-3-245:/data/storage07,ip-10-0-3-162:/data/storage07
Status: Available
Claims:
Pods:
```

**Comment by dkoshkin** [comment]
Another one
```
Name:        kismatic-test-volume
Capacity:    1.00GB
Available:   1.00GB
Available:   1
Available:   1
Bricks:      ip-10-0-3-211:/data/kismatic-test-volume
Status:      Bound
Claims:      default/claim1
Pods:        default/mypod1,default/mypod3

Name:        storage01
Capacity:    2.00GB
Available:   2.00GB
Available:   2
Available:   1
Bricks:      ip-10-0-3-211:/data/storage01,ip-10-0-3-245:/data/storage01
Status:      Available
Claims:
Pods:

Name:        storage02
Capacity:    2.00GB
Available:   2.00GB
Available:   1
Available:   2
Bricks:      ip-10-0-3-211:/data/storage02,ip-10-0-3-162:/data/storage02
Status:      Available
Claims:
Pods:

Name:        storage03
Capacity:    1.00GB
Available:   1.00GB
Available:   2
Available:   2
Bricks:      ip-10-0-3-245:/data/storage03,ip-10-0-3-162:/data/storage03,ip-10-0-3-211:/data/storage03,ip-10-0-3-48:/data/storage03
Status:      Bound
Claims:      default/claim2
Pods:        default/mypod2,default/mypod4

Name:        storage04
Capacity:    2.00GB
Available:   2.00GB
Available:   4
Available:   1
Bricks:      ip-10-0-3-245:/data/storage04,ip-10-0-3-211:/data/storage04,ip-10-0-3-162:/data/storage04,ip-10-0-3-48:/data/storage04
Status:      Bound
Claims:      default/my-claim3
Pods:

Name:        storage05
Capacity:    2.00GB
Available:   2.00GB
Available:   1
Available:   4
Bricks:      ip-10-0-3-211:/data/storage05,ip-10-0-3-162:/data/storage05,ip-10-0-3-245:/data/storage05,ip-10-0-3-48:/data/storage05
Status:      Available
Claims:
Pods:

Name:        storage06
Capacity:    2.00GB
Available:   2.00GB
Available:   1
Available:   3
Bricks:      ip-10-0-3-211:/data/storage06,ip-10-0-3-245:/data/storage06,ip-10-0-3-162:/data/storage06
Status:      Available
Claims:
Pods:

Name:        storage07
Capacity:    2.00GB
Available:   2.00GB
Available:   3
Available:   1
Bricks:      ip-10-0-3-211:/data/storage07,ip-10-0-3-245:/data/storage07,ip-10-0-3-162:/data/storage07
Status:      Available
Claims:
Pods:
```

**Comment by dkoshkin** [comment]
Added `Containers` and `MountPath`
```
➜  out git:(master) ✗ ./kismatic volume list
Name:                kismatic-test-volume
Capacity:            1.00GB
Available:           1.00GB
Available:           1
Available:           1
Bricks:              ip-10-0-3-211:/data/kismatic-test-volume
Status:              Bound
Claim:               default/claim1
Pods:
  default/mypod1
    Containers:
      myfrontend1
        MountPath:   /var/www/html
  default/mypod3
    Containers:
      myfrontend3
        MountPath:   /var/www/html


Name:        storage01
Capacity:    2.00GB
Available:   2.00GB
Available:   2
Available:   1
Bricks:      ip-10-0-3-211:/data/storage01,ip-10-0-3-245:/data/storage01
Status:      Available
Claim:
Pods:


Name:        storage02
Capacity:    2.00GB
Available:   2.00GB
Available:   1
Available:   2
Bricks:      ip-10-0-3-211:/data/storage02,ip-10-0-3-162:/data/storage02
Status:      Available
Claim:
Pods:


Name:                storage03
Capacity:            1.00GB
Available:           1.00GB
Available:           2
Available:           2
Bricks:              ip-10-0-3-245:/data/storage03,ip-10-0-3-162:/data/storage03,ip-10-0-3-211:/data/storage03,ip-10-0-3-48:/data/storage03
Status:              Bound
Claim:               default/claim2
Pods:
  default/mypod2
    Containers:
      myfrontend2
        MountPath:   /var/www/html
  default/mypod4
    Containers:
      myfrontend4
        MountPath:   /var/www/html
```
